### ISSUE TITLE
Cannot reproduce the result

### ISSUE THREAD

**Issue by TrueNobility303** [hao-ai-lab/Consistency_LLM#13]
Dear authors,

I train the CLLM model on GSM8k with Abel-7B-001 as the teacher model, using the dataset
cleaned_gsm8k_jacobi dataset you provided on huggingface, and run the train_cllm.sh, and set "use_gt_labels" in the file train_cllm_global should be set to False according to [this previous issue](https://github.com/hao-ai-lab/Consistency_LLM/issues/11).

The trained model only has an accuracy 0.44 after running bash eval/gsm8k/acc.sh, which is much lower than the result of the checkpoint provided by you.

Could you tell me what is wrong? What is the exact hyperparameter to reproduce the results?

I would greatly appreciate it if you could help me.

Best regards.

**Comment by karrykkk** [comment]
Hi~ Thanks for your interest in our work! ☺️
Please set `use_gt_labels` as True for more strict AR labels(as `teacher_output_ids` may not be accurate though closer to origin distribution) and train a whole epoch with [this dataset](https://huggingface.co/datasets/cllm/cleaned_gsm8k_jacobi_max_new_tokens16_augTrue_labels_True_max_seq_len_256/tree/main) (`max_new_seq_len=256` in the collected Jacobi trajectory). Model collapse may happen during training but the AR loss of ground-truth labels will pull the model back to the normal distribution after more training.
Feel free to reach out if there are still problems.

**Comment by TrueNobility303** [comment]
I use exactly the same dataset and train for a whole epoch, but no matter whether setting use_gt_labels as True/ False can not have the desired result.

**Comment by karrykkk** [comment]
Hi~ Sorry for the confusion. This may result from a bug (i.e. forgetting to add `detach()` to `loss` which leads to extra backpropagation) in the training script. We have fixed this issue in the latest version of the code: https://github.com/hao-ai-lab/Consistency_LLM/blob/425691eef0eb29d686c7cb351eb1d07269d51b32/cllm/cllm_trainer_global.py#L102
You can try running the updated code and the results should be normal now. Let me know if you have any other questions!

**Comment by TrueNobility303** [comment]
Thanks for your reply. I will keep trying!

**Comment by karrykkk** [comment]
My bad😥... While we do use `.detach()` during CLLM training, we found the key bug causing this result is that `labels_ids` in [the dataset we provided before](https://huggingface.co/datasets/cllm/cleaned_gsm8k_jacobi_max_new_tokens16_augTrue_labels_True_max_seq_len_256/tree/main). As shown in the screenshot below, the labels_ids only include the question, and the ground-truth answer is missing.
![image](https://github.com/hao-ai-lab/Consistency_LLM/assets/113419191/23bf248c-4970-4984-92e5-c685f326173c) 
We have launched a modified version and will update the dataset in the hugging face as soon as the generation ends.  You can either wait for the generation to complete or generate your own dataset by running the provided script. Sorry again and thanks for your patience and understanding.

**Comment by TrueNobility303** [comment]
But it is strange that setting use_gt_labels=False still does not solve this problem.

**Comment by snyhlxde1** [comment]
Hi @TrueNobility303. Thanks for your patience! We have identified the problems in the training script:

1. instead of using `complete_teacher_output_ids`, `teacher_output_ids` from the trajectory dataset should be used.
2. during training, model should be loaded in bf16 (in consistency with the setting when generating the trajectories). 

After the patches, after training the model, you should be able to see a much better performance as we have reported in the paper.

**Comment by Songwxuan** [comment]
Hi~Thank you for your reply. So after the modification, should I set use_gt_labels as True or not?

**Comment by karrykkk** [comment]
Sorry for the earlier confusion. We have checked that the current version, with `use_gt_labels` set to `False`, can reproduce the results reported in the paper.

**Comment by Songwxuan** [comment]
Thanks a lot!

**Comment by Songwxuan** [comment]
Hi @karrykkk @snyhlxde1 , sorry for a further question. Because I have deployed CLLM on a new work but got a relative low accuracy, I would like to ask what will happen if I set use_gt_labels = True. In my opinion, this will lead to higher accuracy with some weakness as well.
